### PR TITLE
Wednesday Hardening!

### PR DESCRIPTION
**Pull request by teryk**
- Http error 400 is sent when trying an incorrect issue transition
- Validate a 400 is sent when WS issues/search cannot parse the date
- revent deletion of the last admin group when no standalone admin user

### CODE REVIEW

File: server/sonar-server/src/test/java/org/sonar/server/issue/ws/SearchActionMediumTest.java [hunk 1]
```diff
@@ -609,14 +615,24 @@ public void display_deprecated_debt_fields() throws Exception {
...
+    WsTester.Result result = wsTester.newGetRequest(API_ENDPOINT, SEARCH_ACTION)
       .setParam("resolved", "false")
       .setParam(WebService.Param.FACETS, "severities")
       .setParam("facetMode", DEPRECATED_FACET_MODE_DEBT)
       .execute();
     result.assertJson(this.getClass(), "display_deprecated_debt_fields.json");
   }
 
+  @Test
+  public void fail_when_invalid_format() throws Exception {
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/issue/ws/SearchActionMediumTest.java]: Next time, please do not add tests in class using ServerTester but create another test class that's using EsTester and DbTester

File: server/sonar-server/src/main/java/org/sonar/server/issue/workflow/State.java [hunk 2]
```diff
@@ -43,47 +44,46 @@ public State(String key, Transition[] outTransitions) {
   }
 
   private static void checkDuplications(Transition[] transitions, String stateKey) {
-    Set<String> keys = Sets.newHashSet();
-    for (Transition transition : transitions) {
-      if (keys.contains(transition.key())) {
+    Set<String> keys = new HashSet<>();
+
+    Arrays.stream(transitions)
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/issue/workflow/State.java]: 👍

File: server/sonar-server/src/main/java/org/sonar/server/issue/workflow/State.java [hunk 3]
```diff
@@ -43,47 +44,46 @@ public State(String key, Transition[] outTransitions) {
...
+      .filter(transition -> !keys.add(transition.key()))
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/issue/workflow/State.java]: Strange way to collect key in keys...

Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/issue/workflow/State.java]: I understand. With the lack of accumulator, it's hard to be concise. It's good to use the return value of the `Set.add` method to check for a duplication

**Pull request merged**